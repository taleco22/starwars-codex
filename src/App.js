import React from "react";
import {BrowserRouter, Route, Switch} from "react-router-dom";
import lightsaberMP3 from "./audio/sfx/lightsaber.mp3";
import lightsaberOGG from "./audio/sfx/lightsaber.ogg";
import mp3 from "./audio/starwars.mp3";
import ogg from "./audio/starwars.ogg";
import Categories from "./Routes/Categories";
import Home from "./Routes/Home";
import Subject from "./Routes/Subject";
import "./styles/css/index.css";
import Stars from "./styles/styled/Stars";
import lightning from "./video/lightning.mp4";

function App() {
  let isLoaded = false;
  window.onload = () => {
    isLoaded = true;
  };

  function prepareNavigation(isLoaded) {
    let actual;
    let btn;
    let intervalId;
    intervalId = setInterval(startNavigation, 200);

    function startNavigation(isLoaded) {
      if (isLoaded === false) return;
      clearInterval(intervalId);
      btn = [];
      actual = 0;
      btn = document.querySelectorAll("button");
      if (btn.length === 0) return prepareNavigation(isLoaded);
      document.addEventListener("keyup", applyFocus);
      btn.forEach((a) => a.addEventListener("pointerenter", applyFocus));
      btn.forEach((a) => a.addEventListener("pointerleave", applyFocus));
      btn[0].focus();
      applyFocus();
    }

    function applyFocus(e = {}) {
      if ((e.keyCode === 38) & (actual > 0)) {
        actual -= 1;
        btn[actual].focus();
      } else if ((e.keyCode === 40) & (actual < btn.length - 1)) {
        actual += 1;
        btn[actual].focus();
      } else if (e.target) {
        e.target.focus();
      }
      if (document.activeElement.nextSibling !== null) {
        btn.forEach((v) => {
          v.nextSibling && v.nextSibling.removeAttribute("data-openpreview");
        });
        document.activeElement.nextSibling.setAttribute("data-openpreview", true);
      }
    }
  }

  function start() {
    const audio = document.querySelector("#start");
    audio.load();
    audio.play();

    const video = document.querySelector("#video");
    video.classList.add("play");
    video.load();
    video.play();
    video.addEventListener("ended", () => video.classList.remove("play"));
  }

  return (
    <BrowserRouter>
      <audio autoPlay loop id="audio">
        <source id="ogg" src={ogg} type="audio/ogg" />
        <source id="mp3" src={mp3} type="audio/mpeg" />
      </audio>

      <audio id="start">
        <source id="ogg" src={lightsaberOGG} type="audio/ogg" />
        <source id="mp3" src={lightsaberMP3} type="audio/mpeg" />
      </audio>

      <video id="video">
        <source src={lightning} type='video/mp4; codecs="avc1.42E01E, mp4a.40.2"' />
      </video>

      <Stars />

      <Switch>
        <Route
          exact
          path="/"
          render={(props) => <Home {...props} start={start} fn={prepareNavigation(isLoaded)} />}
        />
        <Route
          exact
          path="/categories"
          render={(props) => <Categories {...props} fn={prepareNavigation(isLoaded)} />}
        />
        <Route
          exact
          path="/categories/characters"
          render={(props) => <Subject {...props} theme="people" fn={prepareNavigation(isLoaded)} />}
        />
        <Route
          exact
          path="/categories/starships"
          render={(props) => (
            <Subject {...props} theme="starships" fn={prepareNavigation(isLoaded)} />
          )}
        />
        <Route
          exact
          path="/categories/films"
          render={(props) => <Subject {...props} theme="films" fn={prepareNavigation(isLoaded)} />}
        />
        <Route
          exact
          path="/categories/vehicles"
          render={(props) => (
            <Subject {...props} theme="vehicles" fn={prepareNavigation(isLoaded)} />
          )}
        />
        <Route
          exact
          path="/categories/species"
          render={(props) => (
            <Subject {...props} theme="species" fn={prepareNavigation(isLoaded)} />
          )}
        />
        <Route
          exact
          path="/categories/planets"
          render={(props) => (
            <Subject {...props} theme="planets" fn={prepareNavigation(isLoaded)} />
          )}
        />
      </Switch>
    </BrowserRouter>
  );
}

export default App;
