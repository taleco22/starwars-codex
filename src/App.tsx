import React from "react";
import { Provider } from "react-redux";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { audio, video } from "./assets";
import { Footer, Stars } from "./components/";
import store from "./pages/Subject/store";
import { Spinner } from "./pages/Subject/styled";

const Categories = React.lazy(() => import("./pages/Categories/Categories"));
const Home = React.lazy(() => import("./pages/Home/Home"));
const Subject = React.lazy(() => import("./pages/Subject/Subject"));

const App: React.FC = () => {
  React.useEffect(() => {
    const soundTrack: HTMLMediaElement | null = document.querySelector("#theme");
    soundTrack?.load();
    soundTrack?.play();
  }, []);

  return (
    <BrowserRouter>
      <audio loop={true} id="theme">
        <source id="ogg" src={audio.ogg} type="audio/ogg" />
        <source id="mp3" src={audio.mp3} type="audio/mpeg" />
      </audio>

      <audio id="lightsaberSound">
        <source id="ogg" src={audio.lightsaberOGG} type="audio/ogg" />
        <source id="mp3" src={audio.lightsaberMP3} type="audio/mpeg" />
      </audio>

      <video id="lightsaberVideo">
        <source src={video.lightning} type='video/mp4; codecs="avc1.42E01E, mp4a.40.2"' />
      </video>

      <Stars />

      <Switch>
        <React.Suspense fallback={<Spinner />}>
          <Route exact path="/" component={Home} />
          <Route exact path="/categories" component={Categories} />
          <Provider store={store}>
            <Route exact path="/categories/:theme" component={Subject} />
          </Provider>
        </React.Suspense>
      </Switch>

      <Footer />
    </BrowserRouter>
  );
};

export default App;
