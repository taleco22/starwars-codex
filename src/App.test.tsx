import { shallow } from "enzyme";
import React from "react";
import App from "./App";
import "./setupTest";

describe("<App />", () => {
  it("should render", () => {
    const component = shallow(<App />);
    expect(component.name()).toStrictEqual("BrowserRouter");
  });
});
