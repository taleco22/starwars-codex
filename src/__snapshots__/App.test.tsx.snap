// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render 1`] = `
.c2 {
  display: none;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  color: #ccc;
  font-size: 1.2rem;
}

.c5 {
  border: 1px solid #333;
  border-radius: 12%;
  padding: 0 0.3rem;
  margin: 0 1rem;
  width: 3.6rem;
}

.c5 img {
  width: 100%;
}

.c0 {
  position: fixed;
  top: -100vh;
  border-radius: 50%;
  -webkit-animation-name: hKrxre;
  animation-name: hKrxre;
  -webkit-animation-timing-function: linear;
  animation-timing-function: linear;
  -webkit-animation-iteration-count: infinite;
  animation-iteration-count: infinite;
}

.c1 {
  -webkit-animation: iXyqho 1s linear infinite;
  animation: iXyqho 1s linear infinite;
  -webkit-transform: translateZ(0);
  -ms-transform: translateZ(0);
  transform: translateZ(0);
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  margin: auto;
  z-index: 2;
  border-top: 2px solid grey;
  border-right: 2px solid grey;
  border-bottom: 2px solid grey;
  border-left: 4px solid #ffe81f;
  background: transparent;
  width: 24px;
  height: 24px;
  border-radius: 50%;
}

@media (min-width:768px) {
  .c2 {
    width: 100%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    position: absolute;
    bottom: 0;
    background: rgba(255,255,255,0.03);
    display: block;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
    height: 7rem;
  }
}

@media (min-width:768px) {
  .c3 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    max-width: 60rem;
    width: 40%;
    position: absolute;
    margin: auto;
    left: 0;
    right: 0;
    top: 0;
    bottom: 0;
  }
}

<div>
  <audio
    id="theme"
    loop=""
  >
    <source
      src="starwars.ogg"
      type="audio/ogg"
    />
    <source
      src="starwars.mp3"
      type="audio/mpeg"
    />
  </audio>
  <audio
    id="lightsaberSound"
  >
    <source
      src="lightsaber.ogg"
      type="audio/ogg"
    />
    <source
      src="lightsaber.mp3"
      type="audio/mpeg"
    />
  </audio>
  <video
    id="lightsaberVideo"
  >
    <source
      src="lightning.mp4"
      type="video/mp4; codecs=\\"avc1.42E01E, mp4a.40.2\\""
    />
  </video>
  <audio
    id="lightsaberMove"
  >
    <source
      type="audio/ogg"
    />
    <source
      type="audio/mpeg"
    />
  </audio>
  <div
    class="c0 layer1"
  />
  <div
    class="c0 layer1 a"
  />
  <div
    class="c0 layer2"
  />
  <div
    class="c0 layer2 a"
  />
  <div
    class="c0 layer3"
  />
  <div
    class="c0 layer3 a"
  />
  <div
    class="c1"
  />
  <div
    class="c2"
  >
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <img
            alt="up-arrow"
            src="[object Object]"
          />
        </div>
        - Up
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <img
            alt="down-arrow"
            src="[object Object]"
          />
        </div>
        - Down
      </div>
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          <img
            alt="enter-key"
            src="[object Object]"
          />
        </div>
        - Confirm
      </div>
    </div>
  </div>
</div>
`;
